Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,391 @@
+                else if(integrity < 75)
+                    damage = Math.Clamp((component.Power * 0.002f) * SupermatterComponent.DamageIncreaseMultiplier, 0, SupermatterComponent.MaxSpaceExposureDamage);
+            } //*/
+
+            //if in space
+            if((int)component.Owner.Transform.GridID == 0)
+            {
+                damage = Math.Max((component.Power / 1000) * SupermatterComponent.DamageIncreaseMultiplier, 0.1f);
+            }
+
+            damage = Math.Min(component.DamageArchived + (SupermatterComponent.DamageHardcap * 900f), damage);
+
+            //damage = ((decimal)damage);
+
+            var damageDelta = new DamageSpecifier(_prototypeManager.Index<DamageTypePrototype>("Shock"), (Shared.FixedPoint.FixedPoint2)damage);
+            _damageable.TryChangeDamage(component.OwnerUid, damageDelta, true);
+
+            //entity.GetComponent<DamageableComponent>().TotalDamage = damage;
+
+            //_mapManager.GetGrid(component.Owner.Transform.GridID).TryGetTileRef(component.Owner.Transform.MapPosition.Position, out var TileRef)
+            /*
+            var splodeTime = 180f;
+            var hurtTime = 120f;
+            if (damage >= 900)
+            {
+                Delamination(component, entity, frameTime);
+                entity.SpawnExplosion(50, 50, 50, 75); //TODO: Tune This once splosions work:tm:
+                entity.QueueDelete();
+                return;
+            }
+
+            component.splodeTimer += frameTime;
+            while (damage >= 700 & component.splodeTimer > splodeTime)
+            {
+                component.splodeTimer -= splodeTime;
+                component.emergency_point = true;
+                chat.DispatchStationAnnouncement("CRYSTAL DELAMINATION IMMINENT.", "Supermatter");
+                //TODO: yell over comms about blowing the fuck up
+            }
+
+            component.hurtTimer += frameTime;
+            while (damage >= 50 & damage <= 700 & component.hurtTimer > hurtTime)
+            {
+                component.hurtTimer -= hurtTime;
+                component.warning_point = true;
+                chat.DispatchStationAnnouncement("Danger! Crystal hyperstructure integrity faltering!", "Supermatter");
+                //TODO: yell over comms about being bullied
+            }
+            */
+
+
+
+        }
+
+        public float GetIntegrity(SupermatterComponent component)
+        {
+            var damage = component.Owner.GetComponent<DamageableComponent>().TotalDamage;
+            var integrity = (double) damage / 900;
+            integrity = Math.Round(integrity);
+            integrity = integrity < 0 ? 0 : integrity;
+            return (float) integrity;
+        }
+
+        public void Delamination(SupermatterComponent component, IEntity entity, float frameTime)
+        {
+
+        }
```

Comment:
Resolves, also given this is empty at least leave a comment explaining why?
Also use `EntityUid`